File: 1354-find-players-with-zero-or-one-losses/find-players-with-zero-or-one-losses.cpp
class Solution {
public:
    vector<vector<int>> findWinners(vector<vector<int>>& matches) {
        unordered_map<int,int> playerLost;
        for(int i=0;i<matches.size();++i){
            if(!playerLost.count(matches[i][0])) playerLost[matches[i][0]]=0;
            playerLost[matches[i][1]]--;
        }
        vector<int>allWin,oneLose;

        for(const auto& player:playerLost){
            if(player.second==0) allWin.push_back(player.first);
            else if(player.second==-1) oneLose.push_back(player.first);
        }

        sort(allWin.begin(),allWin.end());
        sort(oneLose.begin(),oneLose.end());
        return {allWin,oneLose};
    }
};

File: 1354-find-players-with-zero-or-one-losses/find-players-with-zero-or-one-losses.py
class Solution(object):
    def findWinners(self, matches):
        """
        :type matches: List[List[int]]
        :rtype: List[List[int]]
        """
        matchesLost = {}

        for win, lose in matches:
            if win not in matchesLost:
                matchesLost[win] = 0
            if lose not in matchesLost:
                matchesLost[lose] = 0
            matchesLost[lose] += 1

        allWin = []
        oneLose = []

        for player, losses in matchesLost.items():
            if losses == 0:
                allWin.append(player)
            elif losses == 1:
                oneLose.append(player)

        return [sorted(allWin), sorted(oneLose)]